extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileWriter.java

```diff
@@ -54,67 +51,70 @@ public ParquetFileWriter(
         this.targetPageSize = targetPageSize;
         this.allocator = allocator;
         this.extraMetaData = new HashMap<>(extraMetaData);
-        writeChannel = channelsProvider.getWriteChannel(filePath, false); // TODO add support for appending
-        writeChannel.write(ByteBuffer.wrap(ParquetFileReader.MAGIC));
+        // TODO add support for appending
+        bufferedOutput = new BufferedStreamOverWriteChannel(channelsProvider.getWriteChannel(filePath, false),
+                OUTPUT_BUFFER_SIZE);
+        bufferedOutput.write(ParquetFileReader.MAGIC);
         this.type = type;
         this.channelsProvider = channelsProvider;
         this.compressorAdapter = DeephavenCompressorAdapterFactory.getInstance().getByName(codecName);
     }
 
+    /**
+     * Following method is unused and untested.
+     */
```

Comment:
Maybe don't JavaDoc that. ;-)

I am 98% sure that this method is for adding a row group to a different file that will be referenced from the footer of the "main" file. Parquet allows this, we technically support reading it, but I've never seen it in the wild. It might be useful for parallel writing. I am suspicious that nothing is recording the offset indices.